Wheel json reader

LanceMatthies opened the pull request:
Fixed old code review, added steering variable to json, and added Simparams.h header file for sim parameter object

LanceMatthies committed:
read wheel and job json for sim

LanceMatthies committed:
fix default value errors for params without default values

LanceMatthies committed:
fix parameter order and include new json values in header file

LanceMatthies committed:
pass wheel json and job json to wheel sim, calls WheelSimulator with wheel object instead of individual wheel parameters

LanceMatthies committed:
add terrain json

LanceMatthies committed:
debug terrain json

LanceMatthies committed:
fix to include Terrain header

LanceMatthies committed:
debug terrain json more

LanceMatthies committed:
debug

LanceMatthies committed:
changed name of slip directory. Removed the date from the name

LanceMatthies committed:
change slip input from json to job script

LanceMatthies committed:
fixed type of slip

LanceMatthies committed:
removed parameterized slip folder name

LanceMatthies committed:
code review changes

LanceMatthies committed:
debug code review

LanceMatthies committed:
more code review debug

LanceMatthies committed:
more debug code review

LanceMatthies committed:
more debug code review

LanceMatthies committed:
add wheel angle in degrees to job json (SimParams.h)

eholandcmu committed:
Merge branch 'main' into wheel-json-reader

eholandcmu commented on include/SimParams.h:
```diff
@@ -0,0 +1,33 @@
+//                     const std::string& output_dir,
+//                     const std::filesystem::path& data_drivepath,
+//                     const json param, float rotational_velocity,
+//                     float step_size, float scale_factor
```
```suggestion
```

eholandcmu commented on include/WheelSimulator.h:
```diff
@@ -23,13 +27,11 @@ class WheelSimulator {
      * @param terrain_filepath Path to the .csv file containing the pre-settled terrain data.
      * @param data_drivepath Path to the data drive
      */
```
Update brief based on new command arguments

eholandcmu commented on include/WheelSimulator.h:
```diff
@@ -43,10 +45,15 @@ class WheelSimulator {
+    // double slip_;
+    // double sim_endtime_;
+    // std::string batch_dir_;
+    // std::string output_dir_;
```
```suggestion
```

eholandcmu commented on include/WheelSimulator.h:
```diff
@@ -43,10 +45,15 @@ class WheelSimulator {
+    // std::string output_dir_;
+    json param_;
+    // float rot_velocity_;
+    // float scale_factor_;
```
```suggestion
```

HelloKitty9292 committed:
added example json files

HelloKitty9292 committed:
update main json paths

HelloKitty9292 committed:
syntax fix

HelloKitty9292 committed:
job_param example json update

HelloKitty9292 committed:
syntax fix

HelloKitty9292 committed:
syntax fix

HelloKitty9292 committed:
updated README

Commit:
wheel simulator minor change

HelloKitty9292 committed:
Merge branch 'wheel-json-reader' of github.com:chrono-cmu/wheel_simulator into wheel-json-reader

eholandcmu commented on WheelSimulator.cpp:
```diff
@@ -146,42 +146,41 @@ void WheelSimulator::ConfigureDEMSolver() {
-    float terrain_density = 2.6e3f;
-    float volume1 = 4.2520508f;
+    float terrain_density = terrain_.terrain_density;
+    float volume1 = terrain_.volume1;
```
It is unnecessary to make these new variables - just call `terrain_.terrain_density`, `terrain_.volume1` etc. whenever you need it.

eholandcmu commented on WheelSimulator.cpp:
```diff
@@ -146,42 +146,41 @@ void WheelSimulator::ConfigureDEMSolver() {
 
 
     // Scale factors
     std::vector<double> scales = {0.0014, 0.00075833, 0.00044, 0.0003, 0.0002, 0.00018333, 0.00017};
```
This really should also be parameterized instead of hardcoded. Add it into the terrain json and struct

eholandcmu commented on include/SimParams.h:
```diff
@@ -0,0 +1,33 @@
+    const std::filesystem::path data_drivepath;
+    float rotational_velocity;
+    float step_size;
+    float scale_factor;
```
This should be a terrain param, not a sim param.

HelloKitty9292 replied:
I assume it’s just the scale factor being terrain parameter?

eholandcmu replied:
Yes

eholandcmu commented on input_files/terrain_parameters.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "world_size_x": 1,
+  "world_size_y": 0.3,
+  "world_size_z": 2,
```
For readability, make these nested params :
```
"world_size" : {
    "x": 1,
    "y", 0.3,
    "z", 2
},
```
This can then be called in the code as 
```
terrain_json["world_size"]["x"];
```

eholandcmu commented on WheelSimulator.cpp:
```diff
@@ -260,7 +259,7 @@ void WheelSimulator::PrepareParticles() {
     // Define simulation parameters
     // TODO: Change this so it isn't hardcoded
-    float total_mass = 4.5f; // kg
+    float total_mass = wheel_.total_mass; // kg
```
It is redundant to define this as a local variable; just call `wheel_.total_mass` where needed

eholandcmu commented on WheelSimulator.cpp:
```diff
@@ -277,7 +276,7 @@ void WheelSimulator::ConfigureWheel() {
 void WheelSimulator::SetupPrescribedMotions() {
     // Families' prescribed motions
-    float w_r = 0.2f;  // TODO: Change this so it isn't hardcoded
+    float w_r = simparams_.rotational_velocity; 
```
again, this is unnecessary, just pull from simparams_ directly where needed.

eholandcmu commented on WheelSimulator.cpp:
```diff
@@ -317,6 +316,10 @@ void WheelSimulator::PerformInitialSink() {
     std::cout << "Setting wheel position" << std::endl;
     if (wheel_tracker_) {
         wheel_tracker_->SetPos(make_float3(-0.25, 0, max_z + 0.01 + wheel_.r_outer));
```
The wheel's xyz position has hardcoded offsets. We should have these as a nested field in sim params, e.g. `"offset": { "x": -0.25, "y":0, "z":0.01}`

eholandcmu commented on WheelSimulator.cpp:
Settling time is hardcoded at 0.5s, add to sim params

eholandcmu commented on WheelSimulator.cpp:
```diff
@@ -42,12 +37,21 @@ WheelSimulator::WheelSimulator(double slip, double sim_endtime, const std::strin
                         {"mu", 0.5},
                         {"Crr", 0.00}
                     })),
```
This should be part of the terrain struct and terrain.json, not hardcoded. You can copy how this was done for the wheel

HelloKitty9292 replied:
Currently the material properties for the wheel is hardcoded inside Wheel.h, should I also move that part to wheel parameters.json and make it configurable? Or is it not supposed to be changed and I can leave it as is?

eholandcmu replied:
Yes, I didn't catch that, it should all be configurable

eholandcmu commented on README.md:
````diff
@@ -7,13 +7,71 @@
 5. Build this code:
 ```bash
 cd build
````
```suggestion
cd build
cmake ..
```
You need to run `cmake ..` whenever CMakeLists.txt changes; it's good habit just to do this every time

eholandcmu reviewed (changes requested):
Cleanup needed, see comments.

chrono-cmu committed:
Wheel-json-reader cleanup, remove any hardcoded parameters in the
original code and replace with json configurable. Removed any unused
variables and updated README.

eholandcmu commented on input_files/wheel_parameters.json:
```diff
@@ -4,5 +4,13 @@
+      "nu": 0.3,            // Poisson's ratio
+      "CoR": 0.3,           // Coefficient of restitution
+      "mu": 0.5,            // Friction coefficient
+      "Crr": 0.0            // Rolling resistance coefficient
```
Json files unfortunately don't allow comments; remove these
